File: back/app/src/main/java/com/schedule/app/dto/ScheduleDTO.java
package com.schedule.app.dto;

import com.schedule.app.enums.ScheduleType;
import java.time.LocalDate;
import java.time.LocalTime;
import lombok.Builder;

@Builder
public record ScheduleDTO(
    int scheduleId,
    LocalDate date,
    LocalTime startTime,
    LocalTime endTime,
    String worktypeName,
    String worktypeColor,
    ScheduleType scheduleType) {}

File: front/src/type/dto/schedule.dto.ts
import { ScheduleEnum } from "@/enum/schedule.enum";

export type ScheduleDTO = {
    scheduleId: string;
    date: string;
    startTime: string;
    endTime: string;
    worktypeName: string;
    worktypeColor: string;
    scheduleType: ScheduleEnum;
};